src/magnify/apps/core/models.py

```diff
@@ -262,8 +288,32 @@ class Meta:
         verbose_name_plural = _("Room user accesses")
 
     def __str__(self):
-        admin_status = " (admin)" if self.is_administrator else ""
-        return f"{capfirst(self.room.name):s} / {capfirst(self.user.name):s}{admin_status:s}"
+        return (
+            f"{capfirst(self.room.name):s} / {capfirst(self.user.name):s} "
+            f"({self.get_role_display():s})"
+        )
+
+    def save(self, *args, **kwargs):
+        """Make sure we keep at least one owner in a room."""
+        if self.pk and self.role != UserRoleChoices.OWNER:
+            accesses = self._meta.model.objects.filter(
+                room=self.room, role=UserRoleChoices.OWNER
+            ).only("pk")
+            if len(accesses) == 1 and accesses[0].pk == self.pk:
+                raise PermissionDenied("A room should keep at least one owner.")
+        return super().save(*args, **kwargs)
```

Comment:
```suggestion
        """Make sure we keep at least one owner in a room."""
        if self.pk and self.role != UserRoleChoices.OWNER:
            accesses = self._meta.model.objects.filter(
                room=self.room, role=UserRoleChoices.OWNER
            ).only("pk")
            if len(accesses) == 1 and accesses[0].pk == self.pk:
                raise PermissionDenied("A room should keep at least one owner.")
        return super().save(*args, **kwargs)
```
?

Reply:
It's tested in the file `test_core_api_room_user_access.py`. 
The test is called: `test_api_room_user_access_delete_owners_last_owner`